packages/devtools/src/docker/types.ts

```diff
@@ -0,0 +1,107 @@
+export type ComposeSpecVersion = '3.9'
+
+export interface ComposeSpecServiceHealthcheck {
+    disable?: boolean
+    interval?: string
+    retries?: number
+    test?: string | string[]
+    timeout?: string
+    start_period?: string
+    start_interval?: string
+}
+
+export interface ComposeSpecServiceBuild {
+    context?: string
+    dockerfile?: string
+    dockerfile_inline?: string
+    args?: ListOrDict
+    ssh?: ListOrDict
+    labels?: ListOrDict
+    cache_from?: string[]
+    cache_to?: string[]
+    no_cache?: boolean
+    network?: string
+    pull?: boolean
+    target?: string
+    privileged?: boolean
+    tags?: string[]
+    platforms?: string[]
+}
+
+export interface ComposeSpecPortDefinition {
+    name?: string
+    mode?: string
+    host_ip?: string
+    target?: number
+    published?: string | number
+    protocol?: string
+}
+
+export type ComposeSpecPortNumber = string | number
+
+export type ComposeSpecPort =
+    | `${ComposeSpecPortNumber}:${ComposeSpecPortNumber}`
+    | ComposeSpecPortNumber
+    | ComposeSpecPortDefinition
+
+export interface ComposeSpecServiceVolumeDefinition {
+    type: string
+    source?: string
+    target?: string
+    read_only?: boolean
+    consistency?: string
+    bind?: {
+        propagation?: string
+        create_host_path?: boolean
+        selinux?: 'z' | 'Z'
+    }
+    volume?: {
+        nocopy?: boolean
+    }
+    tmpfs?: {
+        size?: number | string
+        mode?: number
+    }
+}
+
+export type ComposeSpecServiceVolume = string | ComposeSpecServiceVolumeDefinition
+
+export type ComposeSpecDependencyCondition = 'service_started' | 'service_healthy' | 'service_completed_successfully'
```

Comment:
Would there be any value in making these constants?

Reply:
Autocomplete

<img width="684" alt="Screenshot 2024-03-12 at 1 27 41 PM" src="https://github.com/LayerZero-Labs/devtools/assets/1451480/2f69b2f8-cbf0-4c8e-bfa5-e85caba006b1">